httpserver/service_node.cpp

```diff
@@ -255,13 +261,34 @@ void ServiceNode::save_bulk(const std::vector<Item>& items) {
     reset_listeners();
 }
 
-void ServiceNode::on_swarm_update(const all_swarms_t& all_swarms) {
+void ServiceNode::on_swarm_update(const block_update_t& bu) {
     if (!swarm_) {
-        BOOST_LOG_TRIVIAL(trace) << "initialized our swarm";
+        BOOST_LOG_TRIVIAL(info) << "Initialized our swarm";
         swarm_ = std::make_unique<Swarm>(our_address_);
     }
 
-    const SwarmEvents events = swarm_->update_swarms(all_swarms);
+    if (bu.height != block_height_) {
```

Comment:
I think we can be more restrictive here and only accept blocks that are newer than `block_height_`:
```cpp
if (bu.height > block_height_)
```

Reply:
Changed to compare block hashes instead and accepting blocks lower than the current height (to allow reorgs?).